### ISSUE TITLE
Random false-positive `Variable is never used` when the variable used only in a macro

### ISSUE THREAD

**Issue by vlad20012** [intellij-rust/intellij-rust#9468]
## Environment

* **IntelliJ Rust plugin version:** 0.4.181.SNAPSHOT-222-dev
* **Rust toolchain version:** 1.64.0 (a55dd71d5 2022-09-19) x86_64-unknown-linux-gnu
* **IDE name and version:** IntelliJ IDEA 2022.2.2 Ultimate Edition (IU-222.4167.29)
* **Operating system:** Linux 5.15.0-48-generic
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts

## Problem description

The most obvious manifestation of the bug is randomly appearing `Variable is never used` false-positive:

![image](https://user-images.githubusercontent.com/3221931/194115009-ea8f9451-92bd-404e-95b1-0b118facc633.png)

![image](https://user-images.githubusercontent.com/3221931/194114717-9dab2973-c889-4f26-b3a2-66fb02709065.png)

Other symptoms:
- `Go To Declaration` does not work for `a`
- Find usages for `let a` does not show the usage in the macro call
- But `Show recursive macro expansion` works for the macro call:
![image](https://user-images.githubusercontent.com/3221931/194115318-bb3467b8-ed44-4810-ac8a-df996149aee2.png)

## Steps to reproduce

It reproduces very randomly, in this sense the bug is similar to #9432. I think it's possible to reproduce it this way:
1. Type this code somewhere in the codebase:
```rust
macro_rules! as_is { ($($ t:tt)*) => { $($ t)* } }

// Foo111
fn main() {
    let a = 0;
    as_is! {
        a;
    }
}
```
2. Leave the tab with this code always open.
3. Then don't type. Use navigation to jump over other source files (e.g. stdlib), use navigation, find usages and so on, i.e., navigation actions, without performing code modifications.
4. Return to the sample code periodically and type something in the comment `// Foo111`. Then hope you will see the false-positive there.

Alternatively, you can leave the IDE inactive for several hours, then open it with a hope to with the false-positive there.
